src/styles/main.scss

```diff
@@ -0,0 +1,62 @@
+$day-size: 100px;
+$border-color: #000;
+$default-bg-color: #eee;
+$hover-bg-color: #ffbfcb;
+$font-size: 30px;
+$font-family: Arial, sans-serif;
+$columns: 7;
+$padding: 10px;
+$animation-duration: 0.5s;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: calc(#{$columns} * (#{$day-size} + 1px) - 1px);
```

Comment:
The calculation for the calendar width uses a combination of variables and arithmetic operations. Ensure that the calculation accounts for all necessary spacing and border widths to avoid layout issues.